lib/rules/prefer-lookaround.ts

```diff
@@ -172,6 +185,205 @@ function getSideEffectsWhenReplacingCapturingGroup(
     }
 }
 
+/**
+ * An element of a pattern consisting only of assertions that may be placed before the starting capturing group.
+ * e.g.
+ * /^(foo)bar/ -> ^
+ * /\b(foo)bar/ -> \b
+ * /(?:^|\b)(foo)bar/ -> (?:^|\b)
+ * /(?<=f)(oo)bar/ -> (?<=f)
+ * /(foo)bar/ -> null
+ */
+type LeadingAssertion =
+    | (EdgeAssertion & { kind: "start" })
+    | WordBoundaryAssertion
+    | LookbehindAssertion
+    | (Group & {
+          alternatives: Alternative & { elements: [LeadingAssertion] }
+      })
+/**
+ * An element of a pattern consisting only of assertions that may be placed after the ending capturing group.
+ * e.g.
+ * /foo(bar)$/ -> $
+ * /foo(bar)\b/ -> \b
+ * /foo(bar)(?:\b|$)/ -> (?:\b|$)
+ * /foo(ba)(?=r)/ -> (?=r)
+ * /foo(bar)/ -> null
+ */
+type TrailingAssertion =
+    | (EdgeAssertion & { kind: "end" })
+    | WordBoundaryAssertion
+    | LookaheadAssertion
+    | (Group & {
+          alternatives: Alternative & { elements: [TrailingAssertion] }
+      })
+
+/** Checks whether the given element is leading assertion or not  */
+function isLeadingAssertion(element: Element): element is LeadingAssertion {
+    if (element.type === "Assertion") {
+        return (
+            element.kind === "word" ||
+            element.kind === "start" ||
+            element.kind === "lookbehind"
+        )
+    } else if (element.type === "Group") {
+        if (element.alternatives.length === 0) return false
+        for (const alt of element.alternatives) {
+            if (alt.elements.length !== 1) return false
+            if (!isLeadingAssertion(alt.elements[0])) return false
+        }
+        return true
+    }
+    return false
```

Comment:
Why are there these requirements for leading and trailing assertions? As far as I see it, any zero-length element will work. E.g. `/foo(bar)(?:^|$|a{0}\b|(?=x)|(?<=y))/` -> `/foo(?=bar(?:^|$|a{0}\b)|(?=x)|(?<=y))/` is a valid transformation. Of course, the regex doesn't make sense, but we have other rules that will report these problems.

Reply:
Thank you for your review! 
I didn't come up with an idea to check for zero-length element.
I will change to use that method.